js/pricing.js

```diff
@@ -101,15 +102,29 @@ $(
         /**
          * Submits consent transaction.
          *
-         * <p>Prepares Privacy, Support Agreement consents data and transaction Url for sending and
-         * calls send transaction function.
+         * <p>Prepares transaction Url, Privacy and Support Agreement consents data, generates
+         * the reCAPTCHA token and adds it to the request data. After that calls send transaction
+         * function.
          */
         function submitOrder() {
+            const transactionUrl = `${apiUrl}/transaction`;
             const privacyConsent = $privacyConsent.prop("checked");
             const supportAgreementConsent = $supportAgreementConsent.prop("checked");
-            const data = JSON.stringify({privacyConsent, supportAgreementConsent});
-            const transactionUrl = `${apiUrl}/transaction`;
-            sendPaymentTransaction(transactionUrl, data);
+
+            showRedirect();
+
+            grecaptcha.ready(() => {
+                grecaptcha
+                    .execute(reCaptchaSiteKey, {action: 'submitPaymentTransaction'})
+                    .then((token) => {
+                        const transaction = {
+                            consent: {privacyConsent, supportAgreementConsent}
+                        };
+                        transaction.token = token;
+                        const transactionRequest = JSON.stringify(transaction);
+                        sendPaymentTransaction(transactionUrl, transactionRequest);
```

Comment:
`sendPaymentTransaction` accepts a `TransactionRequest` object, but here you pass a JSON string instead. Don't you have a warning here?
I'd better do the `JSON.stringify` right before sending the actual AJAX request.